### PR TITLE
Migrate DatabaseMetrics to use a labeled metrics factory [PLEN-164]

### PR DESCRIPTION
**Pull request by nicu-da**
This will simplify the db metrics and make them available only when using the prometheus exporter

Metrics being replaced:

<table>
<tr>
	<td> Old metric
	<td> New Metric
<tr>
	<td> daml.{{prefix}}.{{operation}}.wait
	<td> daml.db.wait {name = "daml.{{prefix}}.{{operation}}"}
<tr>
	<td> daml.{{prefix}}.{{operation}}.exec
	<td>daml.db.exec {name = "daml.{{prefix}}.{{operation}}"}
<tr>
	<td> daml.{{prefix}}.{{operation}}.translation
	<td>daml.db.translation {name = "daml.{{prefix}}.{{operation}}"}
<tr>
	<td> daml.{{prefix}}.{{operation}}.compression
	<td>daml.db.compression {name = "daml.{{prefix}}.{{operation}}"}
<tr>
	<td> daml.{{prefix}}.{{operation}}.commit
	<td>daml.db.commit {name = "daml.{{prefix}}.{{operation}}"}
<tr>
	<td> daml.{{prefix}}.{{operation}}.query
	<td>daml.db.query {name = "daml.{{prefix}}.{{operation}}"}
</table>

All the metrics are timers therefore the old type of `Summary` is being replaced with a `Histogram`

### CODE REVIEW

File: ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala [hunk 1]
```diff
@@ -18,10 +21,6 @@ trait TransactionStreamsDbMetrics {
   val prefix: MetricName
   val factory: MetricsFactory
 
-  @MetricDoc.GroupTag(
-    representative = "daml.index.db.flat_transactions_stream.<operation>",
-    groupableClass = classOf[DatabaseMetrics],
-  )
```
Comment (atriantafyllos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala:24]: How will this affect documentation? Previously all variables where grouped under the <operation> wildcard.

Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala:24]: I will have to double check, but the metrics should be unique by name, therefore only one instance will be actually output in the public docs, as expected. I will add the label doc as well.

File: ledger/metrics/src/main/scala/com/daml/metrics/ParallelIndexerMetrics.scala [hunk 2]
```diff
@@ -4,16 +4,22 @@
...
-  val initialization = new DatabaseMetrics(prefix, "initialization", factory)
+class ParallelIndexerMetrics(
+    prefix: MetricName,
+    factory: MetricsFactory,
+    labeledMetricsFactory: LabeledMetricsFactory,
+) {
+  val initialization = new DatabaseMetrics(prefix :+ "initialization", labeledMetricsFactory)
 
   // Number of state updates persisted to the database
   // (after the effect of the corresponding Update is persisted into the database,
```
Comment (miklos-da) [inline ledger/metrics/src/main/scala/com/daml/metrics/ParallelIndexerMetrics.scala:25]: We have a bunch of `executor` variables defined but unused below — can we remove them, please?

**Review by miklos-da — APPROVED** [review]

File: ledger/metrics/src/main/scala/com/daml/metrics/ParallelIndexerMetrics.scala (hunk 2 above)
Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/ParallelIndexerMetrics.scala:25]: One is for docs only, and one is actually used..not sure why IJ marks it as unused.

**Comment by lucb-da** [comment]
Why merging all the information inside one label, and not creating specific labels ?
I don't know what `prefix` represent, but `operation` should be in its own label.
The `daml` part of the name is not really useful, we are already in `daml` metrics.

**Comment by nicu-da** [comment]
> Why merging all the information inside one label, and not creating specific labels ? I don't know what `prefix` represent, but `operation` should be in its own label. The `daml` part of the name is not really useful, we are already in `daml` metrics.

@lucb-da That could be an option, though not sure how helpful, as with the current name you can filter out with basic queries. I wouldn't necessarily be too granular with the labels unless we have a need for it based on usage.

File: ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala (hunk 1 above)
Comment (atriantafyllos-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala:24]: But will all the instances be present at docs (i.e. the names that the label can get)?

Comment (atriantafyllos-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala:24]: image

Comment (nicu-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala:24]: @atriantafyllos-da No, the possible names will not be present but it open for debate as to if they should be present. With the labeled metrics it's really simple to explore the metrics by building ad hoc queries, therefore it's best to use the metrics themselves as a source to browse them.

Comment (atriantafyllos-da) [reply ledger/metrics/src/main/scala/com/daml/metrics/IndexDBMetrics.scala:24]: cc @mziolekda

**Pull request merged**